### PR TITLE
Admin & email setup

### PR DESCRIPTION
**Pull request by malminawi**
What I did? 
1- created a copy of "iteration2-demo" branch and added 
     a. The Faculty page. (from my other branch). 
     b. added some CSS to the faculty, logout, login, and the referrals pages. (Css was taken from David's branch)
     c. added the email feature in the faculty create referrals page (faculty can upload an excel file or single email)
2- Merged my branch with Deja's branch. (frontend changes were merged from Deja's branch except the faculty pages)

### CODE REVIEW

File: auth_backend/modules/superAdmin/urls.py [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+from django.urls import path, include
+from .views import CreateReferralView, UserView, bulk_invite
+from rest_framework.routers import DefaultRouter
 
+# The router enables us to create dynamic urls
+
+router = DefaultRouter()
+
+router.register('users-details', UserView, basename='user-details')
+# That is, if for example added 1 or 2 ,etc.it will give show the student that has the id 1, etc.
```
Comment (malminawi) [inline auth_backend/modules/superAdmin/urls.py:10]: if 1, 2, etc. added to the url **

File: frontend/src/components/elements/Cards/groupCard.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import React, {Component} from 'react';
```
Comment (malminawi) [inline frontend/src/components/elements/Cards/groupCard.js:1]: Page added by Deja

File: frontend/pageDesignspecs/studentPg.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.jumbotron2 {
```
Comment (malminawi) [inline frontend/pageDesignspecs/studentPg.css:1]: page dded by deja

File: frontend/pageDesignspecs/mentorPg.css [hunk 4]
```diff
@@ -0,0 +1,51 @@
+.jumbotron1{
```
Comment (malminawi) [inline frontend/pageDesignspecs/mentorPg.css:1]: page dded by deja

File: frontend/src/pages/Faculty/student/index.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        searchQuery: ''
+    };
+    componentDidMount() {
+        this.getStudentData();
+    }
+
+
+
+
+    getStudentData = () => {
```
Comment (malminawi) [inline frontend/src/pages/Faculty/student/index.js:21]: I will have this query inside services/api.

File: auth_backend/modules/superAdmin/views.py [hunk 6]
```diff
@@ -1,8 +1,23 @@
-from .forms import ReferralCreateForm
+import pandas as pd
```
Comment (NullDefault) [inline auth_backend/modules/superAdmin/views.py:1]: we definitely should not import the entirety of pandas - it's a huge library. Just import the stuff you are using.

File: auth_backend/modules/superAdmin/views.py [hunk 7]
```diff
@@ -1,8 +1,23 @@
...
+    permissions as rest_permissions,
+    pagination as rest_pagination
+)
+
+from auth_backend.modules.common import constants as common_constants
 from auth_backend.modules.common.mixins import LoginRequiredMixin, AdminRequiredMixin
+from auth_backend.modules.user.models import Referral, BaseVologUser
+from auth_backend.modules.user.serializers import UserSerializer
 
+from .forms import ReferralCreateForm
```
Comment (NullDefault) [inline auth_backend/modules/superAdmin/views.py:19]: explicit imports are generally cleaner than relative ones

File: auth_backend/modules/superAdmin/views.py [hunk 8]
```diff
@@ -20,3 +35,47 @@ def get_context_data(self, **kwargs):
         })
         return context
 
+class UserView(rest_viewsets.ModelViewSet):
+    """
+    User list and detial view
```
Comment (NullDefault) [inline auth_backend/modules/superAdmin/views.py:40]: typo - enable spell check so these things are caught for you

File: auth_backend/modules/user/listeners.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
+from auth_backend.modules.common import utils as common_utils
+
+
+def send_invite_mail(sender, instance, created, **kwargs):
```
Comment (NullDefault) [inline auth_backend/modules/user/listeners.py:4]: is this properly working as a listener? I thought the syntax is different - might be wrong tho

File: auth_backend/modules/user/models.py [hunk 10]
```diff
@@ -66,18 +70,23 @@ class BaseVologUser(auth_models.AbstractBaseUser, auth_models.PermissionsMixin,
...
+    email = models.EmailField('Email Address')
     code = models.CharField('Referral Code', max_length=255, unique=True, default=common_utils.generate_referral_code)
     role = models.SmallIntegerField('Role', choices=common_constants.ROLE_CHOICES, null=True)
     is_used = models.BooleanField('Is Used', help_text='Is referral code used', default=False)
 
     def __str__(self):
         return f'{self.code} - {self.get_role_display()}'
+
+
+post_save.connect(send_invite_mail, sender=Referral)
```
Comment (NullDefault) [inline auth_backend/modules/user/models.py:92]: This is where the listener is connected, correct?

File: frontend/src/pages/FacultyPage.js [hunk 11]
```diff
@@ -1,21 +1,27 @@
...
-    this.state = {isLoading: true};
+    this.state = {
+        isLoading: true,
+        page: 0
+    };
   }
     componentDidMount() {
         let com = this
-        let xhttp = new XMLHttpRequest();
+        var xhttp = new XMLHttpRequest();
```
Comment (NullDefault) [inline frontend/src/pages/FacultyPage.js:16]: var is invalid js syntax (use let or const instead)

**Review by NullDefault — CHANGES_REQUESTED** [review]
Great work all around - just some small changes to make before we merge it into master

**Pull request merged**